### ISSUE TITLE
Including any subfolder in movie causes matching to completely fail

### ISSUE THREAD

**Issue by nomandera** [MediaBrowser/Emby#3566]
I have had this independently checked on 4.2.1.0 because it seemed too obvious to be a bug. Sure enough if you have a movie with a subfolder it causes absolute havoc with matching.

I enclose a two fake movies that are all but identical except one has a subfolder. [emby_subfolder_examples.zip](https://github.com/MediaBrowser/Emby/files/3506673/emby_subfolder_examples.zip)

The layout is as follows with emby1 matching as expected and the inclusion of the empty subfolder  in emby2. completely breaking matching
```
├── emby1
│   ├── emby.avi
│   └── movie.nfo
└── emby2
    ├── emby.avi
    ├── movie.nfo
    └── randomfolder
```

These examples use URL movie only nfos as anchors for `Smokey and the Bandit` to bypass the need for name guessing but the results are the same without nfo and well named folders.

You can see its pretty radically different.

![screenshot 2](https://user-images.githubusercontent.com/1133344/63115909-dd8ce780-bf87-11e9-980c-e01a26005436.jpg)

Obviously since its fairly standard to have things like a `subs` folder the impact of this seems significant and as usual once Emby has made the initial incorrect match it is a complete pain to find it and make Emby forget the error and try again.

**Comment by LukePulverenti** [comment]
Expected behavior if your file name is not sufficient to match with. Let me go into detail.

In example 1, the movie is the only item in the folder, therefore it takes ownership of the entire folder, and matching occurs based on the folder name.

In example 2, randomfolder is treated as additional content, therefore the movie does not take full ownership of the folder, and matching is based on movie file name. So in this situation you'll need your movie file name to be named nice and neatly so that it can be plugged into the moviedb search engine and produce a meaningful result.

**Comment by nomandera** [comment]
Surely movie.nfo should always take precedence?

That is how everyone else does it (e.g Kodi) as the idea with URL only nfos is specifically to anchor a tvshow or movie and bypass name guessing and deal with edge cases such as the multiple identically named movies being released in the same year.

**Comment by LukePulverenti** [comment]
Not really because what if you have content inside randomfolder? Other users in that situation will complain that they have videos that aren't showing up.

Now you could say well what if randomfolder is empty, then why not ignore it, and yes that would make sense. We do have other topics about ignroing empty folders.

**Comment by nomandera** [comment]
Struggling to follow why Emby does this differently to everyone else, let me go into more detail:

If you have a movie.nfo it is a definitive, (typically hand crafted or at least externally curated), statement on what the movie is. Ignoring it and do any kind of guessing based on file or folder name does not seem logical.

Now i follow the logic in the scenario you are detailing but these users are not following the Emby standard layout as documented here https://github.com/MediaBrowser/Wiki/wiki/Movie%20naming
which has no mention movies within movies.

I wont argue for a default order matching change that could impact legacy users but how viable would it be to have a tick box that says "prioritize nfo" or "kodi compatibility".

Note: Kodi match order https://kodi.wiki/view/NFO_files/Movies

**Comment by LukePulverenti** [comment]
We do use movie.nfo, when the video is the only item in the folder. When there are multiple items, we don't use movie.nfo, because there's no way to know which item it would belong to. Now granted there's only video, so a pretty good guess could be made, but that's not what it currently does.

**Comment by nomandera** [comment]
What is the way forward then as it stands Kodi is doing a radically better job than Emby in this scenario since it is using the information entered by the user and Emby is guessing.

Short term is there any way I can trick Emby into using movie.nfo rather than, for what I can see, supporting it but never actually using it?

**Comment by LukePulverenti** [comment]
I'll have to think about what the next level of improvements would be. For immediate resolution you'd have to either delete randomfolder, or rename the nfo file to have the same filename as the video file.

**Comment by nomandera** [comment]
Shout if you need a sounding board. I have quite a decent understanding of how the non Emby people do this.

If I read what you have said correctly as long as I have no subfolders then nfo will be used. If this is indeed the case then it is a scriptable fix here to flatten every movie folder. Not ideal but a fix is a fix.

**Comment by LukePulverenti** [comment]
Yes that is correct, except for known movie sub-folders such as extras, trailers, etc. Those are considered owned by the movie and will not be treated as additional content.

**Comment by nomandera** [comment]
In the interim, until a full solution is discussed, could you add a folder to the `Movie Extras` list 

https://github.com/MediaBrowser/Wiki/wiki/Movie%20naming

that is ignored.

This would allow existing broken on disk file organizations to to "fixed" by renaming existing sub-folders which is much simpler and far far less risky that mass merging and deleting sub-folders.

Having an ignore folder type is not a bad thing to have long term anyway.

**Comment by LukePulverenti** [comment]
Resolved in 4.8.0.36+.
